### ISSUE TITLE
okta_app_oauth `issuer_mode` inconsistency

### ISSUE THREAD

**Issue by erelado** [okta/terraform-provider-okta#1118]
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v1.1.9
on windows_amd64

+ provider registry.terraform.io/okta/okta v3.26.0

### Affected Resource(s)

- okta_app_oauth

### Terraform Configuration Files

```hcl
resource "okta_app_oauth" "app1" {
  label                     = "app1"
  type                      = "web"
  grant_types               = ["authorization_code", "refresh_token", "implicit"]
  redirect_uris             = ["https://${local.APP_app1_URI}/api/okta-redirect"]
  post_logout_redirect_uris = ["https://${local.APP_app1_URI}"]
  response_types            = ["code", "id_token"]
  login_uri                 = "https://${local.APP_app1_URI}"
  issuer_mode               = "CUSTOM_URL"
  consent_method            = "TRUSTED"
  groups_claim {
    type        = "FILTER"
    filter_type = "REGEX"
    name        = "groups"
    value       = ".*"
  }
}

resource "okta_app_oauth" "app2" {
  label                     = "app2"
  type                      = "web"
  grant_types               = ["client_credentials", "authorization_code", "refresh_token", "implicit"]
  redirect_uris             = ["https://${local.APP_IDENTITY_URI}", "https://${local.APP_IDENTITY_URI}/external/callback/${lower(var.ACCOUNT_NAME)}"]
  post_logout_redirect_uris = ["https://${local.APP_IDENTITY_URI}", "https://${local.APP_IDENTITY_URI}/external/callback/${lower(var.ACCOUNT_NAME)}"]
  response_types            = ["code", "id_token"]
  issuer_mode               = "CUSTOM_URL"
  refresh_token_rotation    = "STATIC"
  consent_method            = "TRUSTED"
  lifecycle {
    ignore_changes = [groups]
  }
}
resource "okta_app_group_assignment" "everyone_to_app2" {
  app_id   = okta_app_oauth.app2.id
  group_id = data.okta_group.everyone.id
  depends_on = [
    okta_app_oauth.app2
  ]
}
resource "okta_app_oauth_api_scope" "app2_api_scopes" {
  app_id = okta_app_oauth.app2.id
  issuer = "https://${var.CUSTOM_SUBDOMAIN}.${var.CUSTOM_DOMAIN}"
  scopes = ["okta.groups.read"]
  depends_on = [
    okta_app_oauth.app2
  ]
}
```

### Expected Behavior

I'm expecting this code (`issuer_mode` specifically) to select the `"Custom URL"` option for the application, like the one I can create via the Okta Admin Console:
![image](https://user-images.githubusercontent.com/20738487/168998443-d9e54e0c-d948-4315-a6e1-ac6952b8c025.png)

### Actual Behavior

In reality, for `app1`, it stays as `"Okta URL"`:
![image](https://user-images.githubusercontent.com/20738487/168999069-e22968ac-1a8e-4948-b067-524d44ea7bca.png)

That being said, for `app2` it does work.
![image](https://user-images.githubusercontent.com/20738487/168999182-254a041d-e824-4a76-b797-06947099f25b.png)

### Steps to Reproduce

1. `terraform apply`

### References

- [okta_app_oauth](https://registry.terraform.io/providers/okta/okta/latest/docs/resources/app_oauth)

**Comment by monde** [comment]
I'll be looking into this @E-RELevant

**Comment by monde** [comment]
@E-RELevant, double checking you have a custom domain set up correctly for this? I just did a quick fiddle with this in our acceptance testing harness `Error: failed to create OAuth application: the API returned an error: Api validation failed: issuerMode. Causes: errorSummary: issuerMode: 'issuerMode' can not be set to CUSTOM_URL without a custom domain configured.`. Clearly, we don't have a custom domain set up on it. Can you do your apply with TF_LOG=1 and send it to mike.mondragon@okta.com ... I'd like to see how the real call to the API `POST /api/v1/apps?activate=true` and how it  responds.

- https://developer.okta.com/docs/reference/api/apps/#details
- https://developer.okta.com/docs/guides/custom-url-domain/main/

**Comment by erelado** [comment]
Thank you, @monde. I've set `$env:TF_LOG="DEBUG"` and sent the log file to your email adderss.

**Comment by erelado** [comment]
@monde Did you get my email message? Do you have any updates regarding the matter?

**Comment by erelado** [comment]
@monde The problem [was found](https://github.com/okta/terraform-provider-okta/pull/1167), I believe...

**Comment by Page1997** [comment]
@E-RELevant @monde @jpf @bobtfish @isometry

**Comment by monde** [comment]
Addressed this issue in [v3.30.0](https://github.com/okta/terraform-provider-okta/releases/tag/v3.30.0) release. Please reopen if the issue hasn't been fully resolved.
